### PR TITLE
Wsmancli: 2.3.1 - CLI utility for openwsman libs.

### PR DESCRIPTION
**Pull request by jameskyle**
Provides a cli interface for the openwsman libraries on unix/linux systems. (and now osx!)

Hard dependencies on pull requests #32114 #32115

### CODE REVIEW

**Comment by clarkewd** [comment]
fwiw I would like to see these added to homebrew.

**Comment by mikemcquaid** [comment]
@BrewTestBot test this please

**Comment by mikemcquaid** [comment]
@clarkewd The build errors need to be fixed first.

**Comment by clarkewd** [comment]
Thanks @mikemcquaid. I created a [pull request](https://github.com/jameskyle/homebrew-rejected/pull/5) which I believe will fix the tests

**Comment by jameskyle** [comment]
New pull incoming with clark's changes.

**Comment by clarkewd** [comment]
I've just updated the latest pull request based on the failing checks after the recent merge by @jameskyle

**Comment by mikemcquaid** [comment]
Lots of failures (and previous comments) that still need to be addressed. Just FYI, no rush :smile: 

http://bot.brew.sh/job/Homebrew%20Pull%20Requests/19978/

**Comment by clarkewd** [comment]
thanks @mikemcquaid  - doesn't look like the changes have been merged in yet so it looks like the test shows the same as last time. @jameskyle - could you merge the latest pull request when you have a chance?

**Comment by jameskyle** [comment]
I'll knock out the rest of the build tests

**Comment by jameskyle** [comment]
The test currently failing on the build server, passes for me:

    % brew test --verbose openwsman
    Testing openwsman
    ==> openwsmand --config-file=openwsman.conf

**Comment by clarkewd** [comment]
@jameskyle - it also passes for me. I was trying to have it do something other than the `--help` test as [suggested by](https://github.com/jameskyle/homebrew-rejected/pull/5/files#r23105073) @mikemcquaid

**Comment by jameskyle** [comment]
@clarkewd well, I'll switch it back to --help to see if that gets us a clear ticket. The original openwsman.conf was bad, so that was failing. But the current one should work.

**Comment by jameskyle** [comment]
On a side, looks like the --help* flags on this suite of apps returns an error code of 1. I changed them to -q, which prints the version and exits.

**Comment by jameskyle** [comment]
@mikemcquaid 

Same deal. The above failure is failing on the build machines, but the test passes locally for everyone that's installed it.

**Comment by jameskyle** [comment]
Considering the failures cannot be reproduced and I can't find enough information from the build server to figure out why it's failing there, I'm changing the test to 'system "true"' to see if this trivial case passes.

**Comment by jameskyle** [comment]
@mikemcquaid 

Tests work locally, but are broken on the build server. Please advise.

File: Library/Formula/openwsman.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      default_cim_namespace = root/cimv2
+      cim_client_frontend = XML
+      vendor_namespaces = OpenWBEM=http://schema.openwbem.org/wbem/wscim/1/cim-schema/2,Linux=http://sblim.sf.net/wbem/wscim/1/cim-schema/2,OMC=http://schema.omc-project.org/wbem/wscim/1/cim-schema/2,PG=http://schema.openpegasus.org/wbem/wscim/1/cim-schema/2,RCP=http://schema.suse.com/wbem/wscim/1/cim-schema/2,DCIM=http://schemas.dell.com/wbem/wscim/1/cim-schema/2,SPAR=http://schema.unisys.com/wbem/wscim/1/cim-schema/2,SVS=http://schemas.ts.fujitsu.com/wbem/wscim/1/cim-schema/2
+      port = 5989
+      ssl = yes
+      verify_cert = no
+    EOS
+
+    # should be able to open and parse the a config file
+    system "openwsmand", "--config-file=#{testpath}/openwsman.conf"
```
Comment (mistydemeo) [inline Library/Formula/openwsman.rb]: Test needs to contain the path of the executable:

```ruby
system "#{bin}/openwsmand", "--config-file=#{testpath}/openwsman.conf"
```

File: Library/Formula/wsmancli.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "openssl"
+
+  def install
+    system "./bootstrap"
+    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "wsman", "-q"
```
Comment (mistydemeo) [inline Library/Formula/wsmancli.rb]: Here too.

**Comment by jameskyle** [comment]
Thanks! 

changes pushed.

File: Library/Formula/openwsman.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      default_cim_namespace = root/cimv2
+      cim_client_frontend = XML
+      vendor_namespaces = OpenWBEM=http://schema.openwbem.org/wbem/wscim/1/cim-schema/2,Linux=http://sblim.sf.net/wbem/wscim/1/cim-schema/2,OMC=http://schema.omc-project.org/wbem/wscim/1/cim-schema/2,PG=http://schema.openpegasus.org/wbem/wscim/1/cim-schema/2,RCP=http://schema.suse.com/wbem/wscim/1/cim-schema/2,DCIM=http://schemas.dell.com/wbem/wscim/1/cim-schema/2,SPAR=http://schema.unisys.com/wbem/wscim/1/cim-schema/2,SVS=http://schemas.ts.fujitsu.com/wbem/wscim/1/cim-schema/2
+      port = 5989
+      ssl = yes
+      verify_cert = no
+    EOS
+
+    # should be able to open and parse the a config file
+    system "#{bin}/openwsmand", "--config-file=#{testpath}/openwsman.conf"
```
Comment (mistydemeo) [inline Library/Formula/openwsman.rb]: Oops, looks like this is actually in `sbin` - sorry to lead you astray.

**Comment by mistydemeo** [comment]
Merged! Thanks for your patience.

**Comment by clarkewd** [comment]
hooray!

**Comment by jameskyle** [comment]
yay! Thanks @mistydemeo and @mikemcquaid !

**Pull request closed without merging**